### PR TITLE
(IAC-581) V4_CFG_TLS_TRUSTED_CA_CERTS is required for TLS enabled Viya on AWS clusters

### PR DESCRIPTION
**Pull request by dhoucgitter**
Issue Description: When deploying Viya4 to IAC created AWS clusters (in particular those using the sassoftware/viya4-iac-aws software to create the cluster), and when either full-stack or front-door TLS is configured for the deployment, it is mandatory that a value be set for the configuration variable V4_CFG_TLS_TRUSTED_CA_CERTS.

### CODE REVIEW

**Comment by dhoucgitter** [comment]
When I was working on development tests for verifying the logging and monitoring behavior for the OpenSearch update last sprint, I was unable to get the AWS TLS enabled Viya deployment to stabilize. You and I had a chat about the certificate oriented log messages that I noticed in the pods having trouble which led to our discussion that for AWS deployed clusters that have TLS enabled, it was necessary to provide a certificate bundle that contains both the intermediate and root certificates for all AWS Regions obtained from AWS [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) by pointing to them using the V4_CFG_TLS_TRUSTED_CA_CERTS configuration variable. Once I did that, the pods that were failing started as expected.

**Comment by AWSmith0216** [comment]
Were you using an external DB provisioned in AWS?

**Comment by dhoucgitter** [comment]
> Were you using an external DB provisioned in AWS?

Yes, I was allowing IAC to provision an external postgres DB, both in AWS and in Azure.

**Comment by thpang** [comment]
For AWS the certs are needed for the external postgres server; however, I don't believe they are needed if you use the internal database. So that's the focus of my question why are they needed? We document the scenario where the external postgres server needs a cert.

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -187,7 +187,7 @@ Viya 4 supports 2 different types of certificate generators, Cert-manager and op
...
 | V4_CFG_TLS_CERT | Path to ingress certificate file | string | | false | If specified, used instead of cert-manager issued certificates | viya |
 | V4_CFG_TLS_KEY | Path to ingress key file | string | | false | Required when V4_CFG_TLS_CERT is specified | viya |
-| V4_CFG_TLS_TRUSTED_CA_CERTS | Path to directory containing only PEM encoded trusted CA certificates files | string | | false | Required when V4_CFG_TLS_CERT is specified. Must include all the CAs in the trust chain for V4_CFG_TLS_CERT. Can be used with or without V4_CFG_TLS_CERT to specify any additionally trusted CAs  | viya |
+| V4_CFG_TLS_TRUSTED_CA_CERTS | Path to directory containing only PEM encoded trusted CA certificates files | string | | false | Required when V4_CFG_TLS_MODE is set to full-stack or front-door and the deployment target is an IAC created AWS cluster using self-signed certificates and an external postgres database. Required when V4_CFG_TLS_CERT is specified. Must include all the CAs in the trust chain for V4_CFG_TLS_CERT. Can be used with or without V4_CFG_TLS_CERT to specify any additionally trusted CAs  | viya |
```
Comment (thpang) [inline docs/CONFIG-VARS.md:190]: This is even more confusing. Simply state if you're using an external database that has TLS enabled one must add the PEM cert file to this location. This applies to Open Source K8s, any 3rd party Postgres Server. Either list exceptions here as a table or something. I believe Azure and GCP work fine with an external database but AWS does not. Same with Open Source as stated earlier.

**Review by thpang — CHANGES_REQUESTED** [review]
Need clarity.

**Comment by thpang** [comment]
I looked at this yesterday. The table and text just seem to be there, no description. I would just rather have text in the note that says when using an external database on AWS/Open Source please refer to this doc. And then have a doc that outlines what to do in terms of using AWS and Open Source showing how where to get the files and copy those into the correct location. I believe somewhere we have this in the docs, but don't know where.

**Review by thpang — APPROVED** [review]
LGTM

File: docs/user/TrustedCACerts.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Using Trusted CA Certs
+
+## How to obtain an AWS certificate bundle
+
+To obtain a certificate bundle that contains both the intermediate and root certificates for all AWS Regions, you can download the PEM file from https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html  
+
+## How to obtain an Opensource Kubernetes certificate bundle
+
+Need some advice here, Thomas are you able to provide what's required here?
```
Comment (jarpat) [inline docs/user/TrustedCACerts.md:9]:
```suggestion
If performing a Viya 4 deployment against Opensource Kubernetes whose infrastructure was created with the [viya4-iac-k8s](https://github.com/sassoftware/viya4-iac-k8s) project, and you want to get the certificates for the external Postgres database, follow the steps in the "notes" section in the [documentation here](https://github.com/sassoftware/viya4-iac-k8s/blob/main/docs/CONFIG-VARS.md#postgresql-server). You will have the option to either use the system default certificates in which case they will be copied to your workspace for use, or you can provide your own generated certificates.

```

File: docs/CONFIG-VARS.md [hunk 3]
```diff
@@ -187,7 +187,15 @@ Viya 4 supports 2 different types of certificate generators, Cert-manager and op
...
+| V4_CFG_TLS_TRUSTED_CA_CERTS | Path to directory containing only PEM encoded trusted CA certificates files | string | | false | Required when using an external database and TLS is enabled and the deployment target is an IAC created AWS or Open Source Kubernetes cluster. See the [Trusted CA Certs](user/TrustedCACerts.md) for AWS and Open Source Kubernetes cert information. Required when V4_CFG_TLS_CERT is specified. Must include all the CAs in the trust chain for V4_CFG_TLS_CERT. Can be used with or without V4_CFG_TLS_CERT to specify any additionally trusted CAs  | viya |
+| V4_CFG_TLS_DURATION | Certificate time to expiry in hours | int | 17531 | false | See note below | viya |
+| V4_CFG_TLS_ADDITIONAL_SAN_DNS | A space separated list of additional SAN DNS entries that you want added to generated certificates. | string | | false | See note below  | viya |
+| V4_CFG_TLS_ADDITIONAL_SAN_IP | A space separated list of additional SAN IP addresses that you want added to generated certificates. | string | | false | See note below  | viya |
+|
+
+
+| Name | Description | Type | Default | Required | Notes | Tasks |
+| :--- | ---: | ---: | ---: | ---: | ---: | ---: |
```
Comment (jarpat) [inline docs/CONFIG-VARS.md:198]: Lines 191-198 are duplicated.
```suggestion
```

Comment (dhoucgitter) [reply docs/CONFIG-VARS.md]: Good catch.

**Review by jarpat — APPROVED** [review]

**Review by thpang — APPROVED** [review]
LGTM

**Pull request merged**